### PR TITLE
Make the project usable as an npm dependency

### PR DESCRIPTION
**Pull request by dmisdm**
#### This issue is an enhancement request
It would be great if this project could be more easily used as an NPM dependency in other applications.

What i've done:
- Produce build using webpack
- Update package.json with main field
- Made a couple of code changes to fix a few issues

### CODE REVIEW

File: build-output/_multiple-bundles/readium-external-libs.js.bundles.js [hunk 1]
```diff
@@ -1 +0,0 @@
-require.config({bundles: {"readium-external-libs":["IPv6","SecondLevelDomains","URIjs","console_shim","cssom","domReady","eventEmitter","jquery","jquerySizes","punycode","readium_cfi_js","underscore"]}});
```
Comment (danielweck) [inline build-output/_multiple-bundles/readium-external-libs.js.bundles.js]: The pre-built Javascript bundles in the `build-output` folder should not be deleted.

File: js/globalsSetup.js [hunk 2]
```diff
@@ -12,63 +12,10 @@
...
-                    // c. Increase k by 1. 
-                    k++;
-                }
-
-                // 8. Return false
-                return false;
-            }
-        });
-    }
-
```
Comment (danielweck) [inline js/globalsSetup.js]: The above code should not be deleted.

File: js/helpers.js [hunk 3]
```diff
@@ -24,452 +24,448 @@
 //  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
 //  OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
 //  OF THE POSSIBILITY OF SUCH DAMAGE.
-define(["./globals", 'underscore', "jquery", "jquerySizes", "./models/spine_item", 'URIjs'], function(Globals, _, $, JQuerySizes, SpineItem, URI) {
```
Comment (danielweck) [inline js/helpers.js]: It looks like most (all?) of the code changes below relate to automatic code formatting (VSCode I presume?). To avoid confusing the objective of this Pull Request, such changes should be reverted (even if I personally love consistent formatting and wish there was a super-strict lint checker ;)

File: yarn.lock [hunk 4]
```diff
@@ -2,17 +2,292 @@
 # yarn lockfile v1
 
 
+"@mrmlnc/readdir-enhanced@^2.2.1":
```
Comment (danielweck) [inline yarn.lock]: Note that the Yarn lock files are woefully out of date in every Readium JS projects ... we should really remove them as only the NPM lock files are maintained.

File: js/views/reader_view.js [hunk 5]
```diff
@@ -93,7 +93,7 @@ var ReaderView = function (options) {
     }
 
 
-    _needsFixedLayoutScalerWorkAround = options.needsFixedLayoutScalerWorkAround;
+    var _needsFixedLayoutScalerWorkAround = options.needsFixedLayoutScalerWorkAround;
     /**
```
Comment (danielweck) [inline js/views/reader_view.js]: Good catch.

File: js/polyfills.js [hunk 6]
```diff
@@ -0,0 +1,2 @@
+// Copy jquery sizes into source and require - so that it will 100% extend the same jquery instance instead of potentially a different one (due to different versions)
```
Comment (danielweck) [inline js/polyfills.js]: I believe @JCCR addressed this in the Evident Point fork, right? (`package.json` uses `"jquery-sizes": "github:JCCR/jquery-sizes#master"`)

File: .vscode/settings.json [hunk 7]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (danielweck) [inline .vscode/settings.json]: Although I am a VSCode user myself (so are several ReadiumJS core contributors too), I think we need to be careful including a config file that could interfere with developers' individual preferences.

File: js/views/media_overlay_player.js [hunk 8]
```diff
@@ -3,101 +3,132 @@
 //  Created by Boris Schneiderman.
 // Modified by Daniel Weck
 //  Copyright (c) 2014 Readium Foundation and/or its licensees. All rights reserved.
-//
```
Comment (danielweck) [inline js/views/media_overlay_player.js]: Code auto-format should not be committed in this PR (if there is a significant and PR-relevant code change, we need to narrow it down to a palatable number of lines).

File: js/views/media_overlay_data_injector.js [hunk 9]
```diff
@@ -1,422 +1,445 @@
 //  Copyright (c) 2014 Readium Foundation and/or its licensees. All rights reserved.
-//
```
Comment (danielweck) [inline js/views/media_overlay_data_injector.js]: Code auto-format should not be committed in this PR (if there is a significant and PR-relevant code change, we need to narrow it down to a palatable number of lines).

File: js/views/cfi_navigation_logic.js [hunk 10]
```diff
@@ -2,166 +2,184 @@
 //
 //  Created by Boris Schneiderman.
 //  Copyright (c) 2014 Readium Foundation and/or its licensees. All rights reserved.
-//
```
Comment (danielweck) [inline js/views/cfi_navigation_logic.js]: Code auto-format should not be committed in this PR (if there is a significant and PR-relevant code change, we need to narrow it down to a palatable number of lines).

**Comment by danielweck** [comment]
Hi Daniel, thank you very much for your PR! :)
I posted inline comments.

I think @JCCR 's comments would be welcome too, as he implemented the "port" of `readium-cfi-js` to a proper NPM package (not a Git submodule anymore), and he also implemented the UMD bundle (addition to the `build-output` folder).

File: build-output/_multiple-bundles/readium-external-libs.js.bundles.js (hunk 1 above)
Comment (dmisdm) [reply build-output/_multiple-bundles/readium-external-libs.js.bundles.js]: oh yep sorry not intentional!

File: .vscode/settings.json (hunk 7 above)
Comment (dmisdm) [reply .vscode/settings.json]: fair enough! maybe best to gitignore this file then?

File: js/polyfills.js (hunk 6 above)
Comment (dmisdm) [reply js/polyfills.js]: its probably the fact that im using yarn and didn't know that yarn.lock was out of date (saw your other comment).

regardless, i feel what i've done is more reliable, the effective jquery dependency could change at any point and the runtime error may not be picked up before shipping...not sure how good the tests are.

**Comment by dmisdm** [comment]
@danielweck sorry about all the formatting stuff!
I tried to make sure they didn't creep up but my git diff viewer didn't show some whitespace changes that github shows!

Also ill be honest i didn't realise this already produced a referenced (in package.json) bundle, i did this change a while ago for one of my projects. Maybe this whole PR doesn't hold much value if it already works as an npm dependency, in which case i won't be offended if its ditched!

I do however prefer webpack and the easy wins it brings - but this is obviously your call!
